internal/data_sources/physical_port_data_source.go

```diff
@@ -0,0 +1,319 @@
+	physicalPortsTF := []physicalPortHits{}
+
+	// Map response body to model
+	for _, physicalPort := range filterPhysicalPort {
+		// Create a map to hold attributes of each physicalPort
+		physicalPortTF := physicalPortHits{
+			ID:                 types.StringValue(physicalPort.ID.String()),
+			Name:               types.StringValue(physicalPort.Name),
+			AccountID:          types.StringValue(physicalPort.AccountID),
+			AvailableBandwidth: types.Int64Value(int64(physicalPort.Product.Bandwidth)),
+			Product: physicalPortProduct{
+				Provider:  types.StringValue(physicalPort.Product.Provider.String()),
+				Duration:  types.Int64Value(int64(physicalPort.Product.Duration)),
+				Location:  types.StringValue(physicalPort.Product.Location),
+				Bandwidth: types.Int64Value(int64(physicalPort.Product.Bandwidth)),
+				PriceNRC:  types.Int64Value(int64(physicalPort.Product.PriceNRC)),
+				PriceMRC:  types.Int64Value(int64(physicalPort.Product.PriceMRC)),
+				CostNRC:   types.Int64Value(int64(physicalPort.Product.CostNRC)),
+				CostMRC:   types.Int64Value(int64(physicalPort.Product.CostMRC)),
+				SKU:       types.StringValue(physicalPort.Product.SKU),
+			},
+			State:     types.StringValue(physicalPort.State),
+			UsedVLANs: convertToTerraformList(physicalPort.UsedVLANs),
+		}
+		physicalPortsTF = append(physicalPortsTF, physicalPortTF)
+	}
+
+	state := physicalPortDataSourceModel{
+		Ports: physicalPortsTF,
+	}
+
+	// Set state
+	diags := resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Helper function to convert a slice of int64 to a list of Int64 values for Terraform
+func convertToTerraformList(vlans []int64) []types.Int64 {
+	result := make([]types.Int64, len(vlans))
+	for i, vlan := range vlans {
+		result[i] = types.Int64Value(vlan)
+	}
+	return result
+}
+
+func filterObjects(physicalPorts []autonomisdkmodel.PhysicalPort, filters []filter) []autonomisdkmodel.PhysicalPort {
+	var result []autonomisdkmodel.PhysicalPort
+
+	for _, physicalPort := range physicalPorts {
+		matches := true
+		for _, filter := range filters {
+
+			filterValue := make([]types.String, 0, len(filter.Values.Elements()))
+			_ = filter.Values.ElementsAs(context.TODO(), &filterValue, false)
+
+			switch strings.ToLower(filter.Name.ValueString()) {
+			case "name":
+				if !matchesFilter(physicalPort.Name, filter.Operator.ValueString(), filterValue) {
+					matches = false
+				}
+			case "location":
+				if !matchesFilter(physicalPort.Product.Location, filter.Operator.ValueString(), filterValue) {
+					matches = false
+				}
+			case "priceMrc":
+				if !matchesFilter(strconv.Itoa(physicalPort.Product.PriceMRC), filter.Operator.String(), filterValue) {
+					matches = false
+				}
+			case "priceNrc":
+				if !matchesFilter(strconv.Itoa(physicalPort.Product.PriceNRC), filter.Operator.String(), filterValue) {
+					matches = false
+				}
+			case "costMrc":
```

Comment:
Is the cost not what Intercloud pays ? I'm not sure users would want to filter on that. According to me the filter should only be done on name, physical port id,  location and priceMrc

Reply:
I remove them from the schema but maybe we should remove them from other datasource and resources